File: 0532-k-diff-pairs-in-an-array/0532-k-diff-pairs-in-an-array.cpp
class Solution {
public:
    int findPairs(vector<int>& nums, int k) {
        set<pair<int,int>>st; 
        for( int i = 0 ; i< nums.size() ; i++){
            for( int j = i+1 ; j< nums.size() ; j++){
                if( abs( nums[i] - nums[j]) == k ){
                    int a = max(nums[i],nums[j]); 
                    int b= min(nums[i],nums[j]); 

                    st.insert({a,b}); 
                }
            }
        }
        return st.size();
    }
};

File: 0532-k-diff-pairs-in-an-array/0532-k-diff-pairs-in-an-array.java
import java.util.*;

class Solution {
    public int findPairs(int[] nums, int k) {
        Set<List<Integer>> seen = new HashSet<>();

        for (int i = 0; i < nums.length; i++) {
            for (int j = i + 1; j < nums.length; j++) {
                if (Math.abs(nums[i] - nums[j]) == k) {
                    int a = Math.min(nums[i], nums[j]);
                    int b = Math.max(nums[i], nums[j]);
                    seen.add(Arrays.asList(a, b));
                }
            }
        }

        return seen.size();
    }
}
